### PR TITLE
Notify LocalSlamResults subscription ends.

### PR DESCRIPTION
**Pull request by gaschler**
[RFC=0002](https://github.com/googlecartographer/rfcs/blob/master/text/0002-cloud-based-mapping-1.md)
PAIR=cschuet

### CODE REVIEW

File: cartographer_grpc/handlers/receive_local_slam_results_handler.h [hunk 1]
```diff
@@ -39,7 +39,12 @@ class ReceiveLocalSlamResultsHandler
                 request.trajectory_id(),
                 [writer](std::unique_ptr<MapBuilderServer::LocalSlamResult>
                              local_slam_result) {
-                  writer(GenerateResponse(std::move(local_slam_result)));
+                  if (local_slam_result) {
```
Comment (cschuet) [inline cartographer_grpc/handlers/receive_local_slam_results_handler.h]: Can you add a comment explaining our convention: empty LocalSlamResult => trajectory finished.

File: cartographer_grpc/map_builder_server.cc [hunk 2]
```diff
@@ -193,4 +198,12 @@ void MapBuilderServer::UnsubscribeLocalSlamResults(
            1u);
 }
 
+void MapBuilderServer::NotifyFinishTrajectory(int trajectory_id) {
+  cartographer::common::MutexLocker locker(&local_slam_subscriptions_lock_);
+  for (auto& entry : local_slam_subscriptions_[trajectory_id]) {
+    LocalSlamSubscriptionCallback callback = entry.second;
+    callback(nullptr);
```
Comment (cschuet) [inline cartographer_grpc/map_builder_server.cc]: Or maybe explain the convention here or in both places. I like comments.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

File: cartographer_grpc/map_builder_server.cc (hunk 2 above)
Comment (gaschler) [reply cartographer_grpc/map_builder_server.cc]: Done

File: cartographer_grpc/handlers/receive_local_slam_results_handler.h (hunk 1 above)
Comment (gaschler) [reply cartographer_grpc/handlers/receive_local_slam_results_handler.h]: Done

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**